notebook/static/notebook/js/notebook.js

```diff
@@ -1960,26 +1960,48 @@ define(function (require) {
         dialog.modal(options.dialog);
         return promise;
     };
-    
+
     /**
-     * Execute or render cell outputs and go into command mode.
+     * Execute cells corresponding to the given indices.
+     *
+     * @param {list} indices - indices of the cells to execute
      */
-    Notebook.prototype.execute_cell = function () {
-        // mode = shift, ctrl, alt
-        var cell = this.get_selected_cell();
-        
-        cell.execute();
+    Notebook.prototype.execute_cells = function (indices) {
+        if (indices.length === 0) {
+            return;
+        }
+
         this.command_mode();
         this.set_dirty(true);
+
+        var cell;
+        for (var i = 0; i < indices.length; i++) {
+            this.select(indices[i]);
```

Comment:
Hmm, yeah, I am not sure that should select each cell as it runs it. Can you explain what the logic is? I am worried that will cause a lot of focus jumping on notebooks with long cells.